app.py

```diff
@@ -397,26 +416,47 @@ def upload_file_gui():
                 continue
 
         # Reset file pointer to the beginning
-        results_df.to_csv(PATH_TO_OUTPUT_CSV, index=False)
+        results_df.to_csv(output_stream, index=False)
+        output_stream.seek(0) 
         try:
             send_results_email(
-             email_recipient, "Disinfo Laundromat Results", "Please find the results from the Disinfo Laundromat analysis attached. ", PATH_TO_OUTPUT_CSV)
+             email_recipient, "Disinfo Laundromat Results", "Please find the results from the Disinfo Laundromat analysis attached. ", io.BytesIO(output_stream.getvalue().encode()), 'laundromat_content_results.csv')
         except Exception as e:
             print(f"Error sending email: {e}, continuing...")
         finally:
-            return send_file( PATH_TO_OUTPUT_CSV, as_attachment=True)
+            bytes_stream = io.BytesIO(output_stream.getvalue().encode())
+            #print("Finally Content Results")
+            return send_file(
+            bytes_stream,
+            mimetype='text/csv',
+            as_attachment=True,
+            download_name='laundromat_content_results.csv'
+        )
 
 #TODO: Figure out a way to API-ify this
 @app.route('/fingerprint-csv', methods=['POST'])
-def fingerprint_file():
+def fingerprint_file_gui():
+    return fingerprint_file(request)
+    # If no file, render the index template
+   # return render_template('index.html', countries=COUNTRIES, languages=LANGUAGES, indicator_metadata=INDICATOR_METADATA)
+
+@app.route('/api/fingerprint-csv', methods=['POST'])
+def fingerprint_file_api():
+    return fingerprint_file(request)
+    #return jsonify({'message': 'File processed successfully'})
+
+def fingerprint_file(request):
     file = request.files['fingerprint-file']
     internal_only = 'internal_only' in request.form
+    email_recipient = request.form.get('email')
+    run_urlscan =  'run_urlscan' in request.form
+    
     if file:    
         df_urls = pd.read_csv(StringIO(file.read().decode('utf-8')))
-        urls = df_urls['Urls'].tolist()  # Assuming 'Urls' is the column name
+        urls = df_urls['url'].tolist()  # Assuming 'Urls' is the column name
 
         # The find_indicators_and_matches function should be defined elsewhere
-        indicators_df, matches_df, indicator_summary, matches_summary = find_indicators_and_matches(urls, internal_only = internal_only)
+        indicators_df, matches_df, indicator_summary, matches_summary = find_indicators_and_matches(urls, internal_only = internal_only, run_urlscan = run_urlscan)
```

Comment:
yes it should! I bet there's behavior here not caught by my latest updates (or possibly even breaking due to them). I'll look into this function, but lets move it somewhere. Maybe `modules.main` to start?